### PR TITLE
Move e2e tests on dymension into the new repo

### PR DESCRIPTION
**Pull request by anhductn2001**
## Description

----

Closes #XXX

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

---;

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

File: .github/workflows/e2e_test.yml [hunk 1]
```diff
@@ -3,32 +3,49 @@ name: E2E Tests
...
+              ghcr.io/dymensionxyz/dymension:e2e
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: e2e
+          path: /tmp/e2e.tar  
+  e2e-tests:
+    needs: build-image
+    uses: dymensionxyz/e2e-tests/.github/workflows/e2e-test-workflow-call.yml@87611e27234bc9b4b7d191f76a546049fa87d39f
```
Comment (pgoos) [inline .github/workflows/e2e_test.yml:49]: can you explain that?
* I don't see `e2e-test-workflow.yml` anywhere
* What is `@87611e27234bc9b4b7d191f76a546049fa87d39f` ?

Comment (anhductn2001) [reply .github/workflows/e2e_test.yml:49]: I have written a reusable workflow here: https://github.com/dymensionxyz/e2e-tests/tree/ducnt/move_tests/.github/workflows

File: .github/workflows/update_docker_images.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# This workflow pushes new dymension images on every commit to main.
```
Comment (pgoos) [inline .github/workflows/update_docker_images.yml:1]: great! 💯

File: .github/workflows/e2e_test.yml (hunk 1 above)
Comment (anhductn2001) [reply .github/workflows/e2e_test.yml:49]: The hash code in this code is the commit hash of the reusable workflow

Comment (anhductn2001) [reply .github/workflows/e2e_test.yml:49]: And we can trigger the reusable workflow from different repositories.

Comment (pgoos) [reply .github/workflows/e2e_test.yml:49]: I get it but once we merge the current PR to default main branch, then your commit hash shouldn't be needed. Can you remove it?

Comment (anhductn2001) [reply .github/workflows/e2e_test.yml]: It has been removed!!

**Review by pgoos — APPROVED** [review]

**Pull request merged**